doc/org/build/gameplay.md

```diff
@@ -0,0 +1,722 @@
+
+We do this by calling `addObject3DComponent` from within an `inflator`.
+
+For example, when a `simpleWater` component is found within a node of a GLTF file, the `inflateSimpleWater` inflator will create a `SimpleWaterMesh` with the given params and associate it with the given `EntityID` :
+
+```typescript
+export function inflateSimpleWater(world: HubsWorld, eid: EntityID, params: SimpleWaterParams) {
+  params = Object.assign({}, DEFAULTS, params);
+  const lowQuality = APP.store.state.preferences.materialQualitySetting !== "high";
+  const simpleWater = new SimpleWaterMesh({ lowQuality });
+  simpleWater.opacity = params.opacity;
+  simpleWater.color.set(params.color);
+  simpleWater.tideHeight = params.tideHeight;
+  simpleWater.tideScale.fromArray(params.tideScale);
+  simpleWater.tideSpeed.fromArray(params.tideSpeed);
+  simpleWater.waveHeight = params.waveHeight;
+  simpleWater.waveScale.fromArray(params.waveScale);
+  simpleWater.waveSpeed.fromArray(params.waveSpeed);
+  simpleWater.ripplesScale = params.ripplesScale;
+  simpleWater.ripplesSpeed = params.ripplesSpeed;
+
+  addObject3DComponent(world, eid, simpleWater);
+  addComponent(world, SimpleWater, eid);
+}
+```
+
+Only one inflator per entity can create and assign an `Object3D`. An `EntityDef` that describes an entity that is both a `SimpleWaterMesh` and a `DirectionalLight` will fail to load:
+
+```typescript
+renderAsEntity(
+  <entity
+      name="Buggy Entity"
+      simpleWater
+      directionalLight
+  />
+); // throws Error("Tried to add an object3D tag to an entity that already has one");
+
+```
+
+By default, if no `inflator` creates an `Object3D` for the entity, then `renderAsEntity` will create and assign it a `Group`.
+
+
+<a id="org7a41123"></a>
+
+### Loading model files
+
+The sections above explain how `renderAsEntity` transforms a `jsx` `EntityDef` into an entity and components, and claims that the process for loading a `gltf` file is equivalent.
+
+We are now ready to understand why, and will explain by example.
+
+The `EntityDef` returned by the `MediaPrefab` template (above) includes a `mediaLoader`, which will cause `renderAsEntity` to invoke the `inflateMediaLoader` inflator and assign a `MediaLoader` component to the entity.
+
+Assume the `MediaPrefab` template is initialized with `src` set to the url of a `gltf` file. In other words, we are trying to use the `MediaPrefab` load a `gltf` file that we can grab and interact with.
+
+We pass the `EntityDef` to `renderAsEntity`, an entity is returned synchronously, and execution continues as normal.
+
+When the `mediaLoading` system runs, it notices the new `MediaLoader` component and starts an asynchronous `coroutine` that determines the media type pointed to by the `src` property, downloads the `gltf` file and loads it as Three.js scene graph via a `GLTFLoader`. Finally, the loaded scene graph is passed into an `EntityDef`&rsquo;s `model` parameter:
```

Comment:
The last sentence might be a bit confusing because reader might think `EntityDef` is a first `EntityDef` that has `mediaLoader` description. Maybe better to clarify that `mediaLoading` system creates a new entity with another EntityDef. (Readers will notice it if they read the following code carefully tho)